src/App.tsx

```diff
@@ -1,61 +1,122 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+const todosWithUsers = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [currentTitle, setCurrentTitle] = useState('');
+  const [currentUserId, setCurrentUserId] = useState(0);
+
+  const [hasNameError, setHasNameError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+
+  const [currentTodoList, setCurrentTodoList] = useState(todosWithUsers);
+
+  function handleTitle(event: React.ChangeEvent<HTMLInputElement>) {
+    setCurrentTitle(event.target.value);
+    setHasNameError(false);
+  }
+
+  function handleUser(event: React.ChangeEvent<HTMLSelectElement>) {
+    setCurrentUserId(+event.target.value);
+    setHasUserError(false);
+  }
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    if (!currentTitle) {
+      setHasNameError(true);
+    }
+
+    if (!currentUserId) {
+      setHasUserError(true);
+    }
+
+    if (currentTitle && currentUserId) {
+      setCurrentTodoList([...currentTodoList, {
+        id: (Math.max(...currentTodoList.map(todo => todo.id)) + 1),
+        title: currentTitle,
+        completed: false,
+        userId: currentUserId,
+        user: usersFromServer.find(user => user.id === currentUserId),
+      }]);
+
+      setCurrentTitle('');
+      setCurrentUserId(0);
+    }
+  }
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={event => handleSubmit(event)}
```

Comment:
You could write like so
```suggestion
        onSubmit={handleSubmit}
```